.github/workflows/publish.yml

```diff
@@ -36,11 +36,28 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: node .__publish__/src/publish/post-workflow-details.js
 
+      # Setting the target repo branch will cause the craft config (.craft.yml) to be taken from the checked out branch
+      # By default, we check out the default branch of the repo.
+      # If you need to maintain diverging craft configs on different branches, add your repo and the merge target branch
+      # (i.e. the branch craft will merge the release branch into) into the if condition below.
+      - name: Set target repo checkout branch
+        # Note: Branches registered here MUST BE protected in the target repo!
+        if: |
+          fromJSON(steps.inputs.outputs.result).repo == 'sentry-javascript' && fromJSON(steps.inputs.outputs.result).merge_target == 'develop' ||
+          fromJSON(steps.inputs.outputs.result).repo == 'sentry-javascript' && fromJSON(steps.inputs.outputs.result).merge_target == 'v7' ||
+          fromJSON(steps.inputs.outputs.result).repo == 'sentry-migr8' && fromJSON(steps.inputs.outputs.result).merge_target == 'main' ||
+          fromJSON(steps.inputs.outputs.result).repo == 'sentry-migr8' && fromJSON(steps.inputs.outputs.result).merge_target == 'tmp-merge-target'
```

Comment:
I opted to temporarily add an entry for [sentry-migr8](https://github.com/getsentry/sentry-migr8) here so that I can test this before we make a new JS SDK release.

Reply:
For sentry-migr8 it should be fine to cut a few releases as it's still just a hackweek project at the moment

Reply:
let's add branch protection to that repo just so we're following the process

Reply:
Done: https://github.com/getsentry/sentry-migr8/settings/branches